client/post-editor/editor-publish-button/index.jsx

```diff
@@ -0,0 +1,127 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { recordEvent } from 'lib/posts/stats';
+import postUtils from 'lib/posts/utils';
+import siteUtils from 'lib/site/utils';
+import Button from 'components/button';
+
+export const getPublishButtonStatus = ( site, post, savedPost ) => {
+	if (
+		postUtils.isPublished( savedPost ) &&
+		! postUtils.isBackDatedPublished( savedPost ) &&
+		! postUtils.isFutureDated( post ) ||
+		(
+			savedPost &&
+			savedPost.status === 'future' &&
+			postUtils.isFutureDated( post )
+		)
+	) {
+		return 'update';
+	}
+
+	if ( postUtils.isFutureDated( post ) ) {
+		return 'schedule';
+	}
+
+	if ( siteUtils.userCan( 'publish_posts', site ) ) {
+		return 'publish';
+	}
+
+	if ( savedPost && savedPost.status === 'pending' ) {
+		return 'update';
+	}
+
+	return 'requestReview';
+};
+
+export default React.createClass( {
```

Comment:
I've made the `localize` change and also changed the component to use the `Component` class. I'll leave the `redux` refactor for future work, if you are ok with that. Looks like a potential rabbit hole :(